webapp/lib/model.dart

```diff
@@ -193,6 +193,11 @@ class User {
   String userEmail;
 }
 
+class ModelDoesNotExistException implements Exception {
+  String cause;
+  ModelDoesNotExistException(this.cause);
+}
```

Comment:
If you're happy with my suggestions above, this would get rewritten as:

```suggestion
class DataModelNotSupported implements Exception {
  String cause;
  DataModelNotSupported(this.cause);
}
```